### ISSUE TITLE
Wrong type of PK in mysql when using tinyint as type

### ISSUE THREAD

**Issue by tomaszwylezek** [mikro-orm/mikro-orm#5996]
### Describe the bug

Driver: mysql
Version: Checked on 6.1.9, 6.3.6 and 6.3.7 and next

When we have two entities like

```
@Entity({ schema: 'foo', tableName: 'with_tiny' })
export class WithTiny {
  @PrimaryKey({ name: 'tiny_id', type: TinyIntType })
  id!: number

  @Property()
  name!: string
}

@Entity({ schema: 'foo', tableName: 'not_tiny' })
export class CasualEntity {
  @PrimaryKey({ name: 'not_tiny_id' })
  id!: number

  @ManyToOne(() => WithTiny, { fieldName: 'tiny_id' })
  tiny!: WithTiny
}
```

Then during initialisation of db it creates them like

```
  CREATE TABLE `with_tiny` (
    `tiny_id` int unsigned NOT NULL AUTO_INCREMENT, <---- NOTE: That should be tinyint
    `name` varchar(255) NOT NULL,
    PRIMARY KEY (`tiny_id`)
  ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci

  CREATE TABLE `not_tiny` (
    `not_tiny_id` int unsigned NOT NULL AUTO_INCREMENT,
    `tiny_id` tinyint unsigned NOT NULL,  <---- NOTE: There is correct one
    PRIMARY KEY (`not_tiny_id`),
    KEY `not_tiny_tiny_id_index` (`tiny_id`)
  ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci
  ```
  
  Where the `with_tiny`.`tiny_id` should be `tinyint` and it's `int` (but in `not_tiny` table the type is correct) what cause the constraint to fail
  
 ```
 DriverException: alter table `foo`.`not_tiny` add constraint `not_tiny_tiny_id_foreign` foreign key (`tiny_id`) references `foo`.`with_tiny` (`tiny_id`) on update cascade; - Referencing column 'tiny_id' and referenced column 'tiny_id' in foreign key constraint 'not_tiny_tiny_id_foreign' are incompatible.
 ```
 
 Tried to find if I'm missing anything but I failed to find any information about it :(

### Reproduction

https://github.com/tomaszwylezek/micro-orm-pk-tinyint
Follow README.md

### What driver are you using?

@mikro-orm/mysql

### MikroORM version

 6.1.9, 6.3.6, 6.3.7 and next

### Node.js version

v18.18.0

### Operating system

mac

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Try this instead:

```ts
@PrimaryKey({ name: 'tiny_id', columnType: 'tinyint' })
id!: number
```

But I can see it won't help, since the support for autoincrement PKs in knex is really weird (and this is a side-effect of how they do things).

**Comment by tomaszwylezek** [comment]
Tried it already - same issue.

**Comment by tomaszwylezek** [comment]
Overall we had similiar issued with AI Pks and `unsigned: false` when generating different entities - we "hacked it" by having separate decorator which is wrapping `@PrimaryKey` and skipping `unsigned | type | autoincrement` for setup process (but ofc it doesn't fix the issue as the types are wrong across the app then as we have the PK type in both entity definition and in places when used as FK)

**Comment by B4nan** [comment]
Fixed via #6065
